src/main/java/Duke.java

```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    int x = Integer.parseInt(n.description.substring(5));
+                    Task edited = ls.remove(x -1);
+                    edited.done();
+                    ls.add(x-1, edited);
+                    for(int i = 1; i <= ls.size(); i++) {
+                        System.out.println(i + " " + ls.get(i -1));
+                    }
+                } else if (n.description.startsWith("unmark")) {
+                    int x = Integer.parseInt(n.description.substring(7));
+                    Task edited = ls.remove(x -1);
+                    edited.undone();
+                    ls.add(x-1, edited);
+                    for(int i = 1; i <= ls.size(); i++) {
+                        System.out.println(i + " " + ls.get(i -1));
+                    }
+                } else if(n.description.startsWith("todo")) {
+                    todo a = new todo(n.description.substring(5));
+                    if(a.description.equals("")) {
+                        throw new DukeException("Nothings here");
+                    } else {
+                        ls.add(a);
+                        System.out.println("Added the following todo");
+                        System.out.println("added: " + a);
+                        System.out.println("New list size is " + ls.size());
+                    }
+                } else if(n.description.startsWith("deadline")) {
+                    String[] parts = n.description.substring(9).split(" /by ");
+                    Deadline d = new Deadline(parts[0], parts[1]);
+                    ls.add(d);
+                    System.out.println("Added the following todo");
+                    System.out.println("added: " + d);
+                    System.out.println("New list size is " + ls.size());
+                } else if(n.description.startsWith("event")) {
+                    String[] parts = n.description.substring(6).split(" /at ");
+                    Event e = new Event(parts[0], parts[1]);
+                    ls.add(e);
+                    System.out.println("added: " + e);
+                    System.out.println("New list size is " + ls.size());
+                } else if(n.description.startsWith("delete")) {
+                    String number = n.description.substring(7);
+                    int x = Integer.parseInt(number);
+                    Task item = ls.remove(x-1);
+                    System.out.println("I've removed the item from the list, stop being so lazy");
+                    System.out.println(item);
+                    System.out.println("you now have " + ls.size() + " things to do");
+                } else {
+                    throw new DukeException("I dont know what you want but this aint it");
+                }
+            } catch (DukeException error){
+                System.out.println("OOPS! " + error.getMessage());
+            } catch (IndexOutOfBoundsException a) {
+                System.out.println("Nothing inside");
+            }
+
+        } 
```

Comment:
This `while` loop tends to become very long and bloated as the number of commands increase. Perhaps consider converting each `if` condition into a command class, and store them into a `HashMap` where you can easily search each command via its trigger word.